lib/item_stream_handler.rb

```diff
@@ -33,9 +33,22 @@ def constrain_size(arr, size)
     arr
   end
 
+  def update_tally_if_necessary
+    if Time.now - ItemTypeTally[:time] >= 24*60*60
```

Comment:
If `ItemTypeTally[:time]` was initialized at 3:45pm (the time the Lambda was first run after deploy) won't this cause a "day"s tallies to begin and end at 3:45 every day? Can we reset `ItemTypeTally` only when the current "day" differs from the "day" of the last written count? May need to store an `:updated_at` in `ItemTypeTally`, updated in `update_count`, which you can inspect in `update_tally_if_necessary` to detect a day rolling over from the last count update..

Reply:
I've changed the method to use differing days. I'm not sure I understand the rest of your comment, but let me know if you think it is not addressed by the current code change